toolkit/tools/graphanalytics/graphanalytics.go

```diff
@@ -0,0 +1,295 @@
+		}
+	}
+
+	printTitle("[DIRECT] Most common unresolved dependencies")
+	printMap(unresolvedPackageDependents, "direct dependents", maxResults)
+
+	return
+}
+
+// printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved direct build requires.
+func printDirectlyClosestToBeingUnblocked(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	srpmsBlockedBy := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllNodes() {
+		if node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgSRPM := filepath.Base(node.SrpmPath)
+
+		dependencies := pkgGraph.From(node.ID())
+		for dependencies.Next() {
+			dependency := dependencies.Node().(*pkggraph.PkgNode)
+			// Only consider unresolved build nodes.
+			if dependency.State != pkggraph.StateBuild && dependency.State != pkggraph.StateUnresolved {
+				continue
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				continue
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+		}
+	}
+
+	printTitle("[DIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "unmet dependencies", maxResults)
+}
+
+// printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved indrect build requires.
+func printIndirectlyClosestToBeingUnblocked(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	srpmsBlockedBy := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllNodes() {
+		if node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgSRPM := filepath.Base(node.SrpmPath)
+
+		search := traverse.BreadthFirst{}
+		search.Walk(pkgGraph, node, func(n graph.Node, d int) (stopSearch bool) {
+			dependency := n.(*pkggraph.PkgNode)
+
+			// Only consider unresolved build nodes.
+			if dependency.State != pkggraph.StateUnresolved && dependency.State != pkggraph.StateBuild {
+				return
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				return
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+
+			return
+		})
+
+	}
+
+	printTitle("[INDIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "total unmet dependencies", maxResults)
+}
+
+// nodeDependencyName returns a common dependency name for a node that will be shared across similair Meta/Run/Build nodes for the same package.
+func nodeDependencyName(node *pkggraph.PkgNode) (name string) {
+	// Prefer the SRPM name if possible, otherwise use the unversioned package name
+	srpmName := filepath.Base(node.SrpmPath)
```

Comment:
>srpmName [](start = 1, length = 8)

Nit: you could drop `srpmName` and rely only on `name`. This also saves 2 lines and the `else` part.:)